tests.py

```diff
@@ -112,6 +138,90 @@ def test_load_ui_returntype():
     app.exit()
 
 
+def test_load_ui_baseinstance():
+    """Tests to see if the baseinstance loading loads widgets on properly"""
+    import sys
+    from Qt import QtWidgets, QtCompat
+    app = QtWidgets.QApplication(sys.argv)
+    win = QtWidgets.QWidget()
+    QtCompat.loadUi(self.ui_qwidget, win)
+    assert hasattr(win, 'lineEdit'), "loadUi could not load instance to win"
+    app.exit()
+
+
+def test_load_ui_signals():
+    """Tests to see if the baseinstance loading loads widgets on properly"""
+    import sys
+    from Qt import QtWidgets, QtCompat
+    app = QtWidgets.QApplication(sys.argv)
+    win = QtWidgets.QWidget()
+    QtCompat.loadUi(self.ui_qwidget, win)
+
+    win.lineEdit.setText('Hello')
+    assert str(win.label.text()) == 'Hello', "lineEdit signal did not fire"
+
+    app.exit()
+
+
+def test_load_ui_invalidpath():
+    """Tests to see if loadUi successfully fails on invalid paths"""
+    import sys
+    from Qt import QtWidgets, QtCompat
+    app = QtWidgets.QApplication(sys.argv)
+    try:
+        QtCompat.loadUi('made/up/path')
+    except IOError:
+        pass  # Success
+    else:
+        raise RuntimeError("Loaded an invalid path successfully")
+    finally:
+        app.exit()
+
+
+def test_load_ui_invalidxml():
+    """Tests to see if loadUi successfully fails on invalid ui files"""
+    import sys
+    invalidXML = os.path.join(self.tempdir, "invalid.ui")
+    with io.open(invalidXML, "w", encoding="utf-8") as f:
+        f.write(u"""
+        <?xml version="1.0" encoding="UTF-8"?>
+        <ui version="4.0" garbage
+        </ui>
+        """)
+
+    from xml.etree import ElementTree
+    from Qt import QtWidgets, QtCompat
+    app = QtWidgets.QApplication(sys.argv)
+    try:
+        QtCompat.loadUi(invalidXML)
+    except ElementTree.ParseError:
+        pass
+    else:
+        raise RuntimeError("Loaded an invalid xml")
+    finally:
+        app.exit()
+
+
+if binding("PySide") or binding("PySide2"):
+    def test_load_ui_overwrite_widget():
+        """Checks to make sure ui file widgets supersede previous ones
+        Unfortunately this test only runs in PySide, because PyQt doesn't
```

Comment:
As funny as it sounds, we need to replicate the PyQt behaviour here.

It's important that what works with PyQt, works with Qt.py is PyQt is really what we are using to guide us. Warts and all. We wouldn't want our users to end up in a situation where "It worked before!" as they were using PySide but switched to PyQt.